content/blog/open-sourcing-workshops-on-demand-part2-deploying-the-backend.md

```diff
@@ -0,0 +1,389 @@
+  * Create a `wod.sh` script in `wod-backend` directory to be used by all other scripts
+  * Source the `wod.sh` file
+  * Setup Ansible-galaxies (`community.general` and `posix`)
+  * Setup Ansible and call the playbook `install_<type>.yml` followed by the `ansible\_check\_<type>.yml`
+
+A﻿ll Playbooks are self documented. Please check for details.
+
+At the end of the installation process:
+
+* you will have a Jupyterhub server running on port 8000
+* You will get a new `wodadmin` user (Default admin)
+* You will get a set of 20 students (Default value)
+
+W﻿e leave it to you to handle the necessary port redirection and SSL certificates management when needed. In our case, we went for a simple yet efficient solution based on an OPNSense Firewall along with a HAProxy setup to manage ports'redirection, HTTP to HTTPS Redirection, SSL Certificates. The backend also includes a Fail2ban service for login security management.
+
+At this point, you should then be able to access your Jupyterhub environment with a few pre-installed set of kernels like `Bash, Python, ansible, ssh, PowerShell`.
+
+Y﻿ou can then start developing new notebooks for your public based environment. And if you don't know how to achieve this, the next article in the series should allow you to learn more about it.
+
+N﻿ow if you are willing to develop private content that cannot be shared because of dedicated IPs with the wider Open Source community, you can read the next part that will explain you how to handle this.
+
+### For private based Workshops-on-Demand (private backend + private workshops on top of default public backend and notebooks)
+
+T﻿he principle remains somehow similar with a few differences explained below.
+
+* Y﻿ou will start by forking the following public private [repo](https://github.com/Workshops-on-Demand/wod-private.git) on github under your own github account (we will refer to it as `Account`)
+* Next, you clone the forked repo
+
+```shellsession
+install$ git clone https://github.com/Account/wod-private.git wod-private
+install$ cd $HOME/wod-private/ansible/group_vars
+```
+
+* Please edit the `all.yml` and `<groupname>` files to customize your setup. T﻿his variable `<groupname>`defines possible backend server in your environement. By default, the project comes with a sample working file named `production` in `ansible/group-vars`. But you could have multiple. In our case, we have defined `sandbox`, `test`, `staging` and several `production` files, all defining a different backend environment. These files will be used to override the default values specified by the public version delivered as part of the default public installation.
+* Commit and push changes to your repo
+* Create an `install.priv` file located in `install` directory when using a private repo (consider looking at [install.repo](https://github.com/Workshops-on-Demand/wod-backend/blob/main/install/install.repo) file for better understanding of the variables.
```

Comment:
This is where the font size gets big. Make sure you have a closed parenthesis after "variables". Add the word "a" prior to "better understanding"